azurerm/resource_arm_managed_disk.go

```diff
@@ -135,7 +138,8 @@ func resourceArmManagedDiskCreate(d *schema.ResourceData, meta interface{}) erro
 		Sku: &compute.DiskSku{
 			Name: (skuName),
 		},
-		Tags: expandedTags,
+		Tags:  expandedTags,
+		Zones: zones,
```

Comment:
Given we set this value always - if a user isn't opted into the Preview most deployments will fail:

```
* azurerm_managed_disk.test: compute.DisksClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="ResourceTypeNotSupportAvailabilityZones" Message="The resource type 'disks' does not support availability zones at location 'westeurope' and api-version '2017-03-30'."
```

Can we instead change this to assign Zones only if the field is set? e.g.

```
if v, ok := d.Get("zones"); ok {
  zones := expandZones(v.([]interface{}))
  createDisk.Zones = zones
}
```

which should remove this error